packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -531,20 +531,45 @@ class XCDevice {
           }
         }
 
+        DeviceConnectionInterface connectionInterface = _interfaceType(device);
+
         String? sdkVersion = _sdkVersion(device);
 
         if (sdkVersion != null) {
           final String? buildVersion = _buildVersion(device);
           if (buildVersion != null) {
             sdkVersion = '$sdkVersion $buildVersion';
           }
+          // Starting in iOS 17, all paired devices are also connected wirelessly.
+          if (sdkVersion.startsWith('17')) {
+            connectionInterface = DeviceConnectionInterface.wireless;
+          }
+        }
+
+        // If a duplicate entry is found in `xcdevice list`, don't overwrite
+        // existing entry when the existing entry indicates the device is
+        // connected and the current entry indicates the device is not connected.
+        // Don't overwrite if current entry's sdkVersion is null.
+        // Don't overwrite if both entries indicate the device is not
+        // connected and the existing entry has a higher sdkVersion.
```

Comment:
maybe explain why we have duplicate in the comment?

Reply:
done